prov/sm2/src/sm2_progress.c

```diff
@@ -42,52 +42,187 @@
 #include "sm2.h"
 #include "sm2_fifo.h"
 
-static inline int sm2_cma_loop(pid_t pid, struct iovec *local,
-			       unsigned long local_cnt, struct iovec *remote,
-			       unsigned long remote_cnt, size_t total,
-			       bool write)
+static int sm2_progress_ipc_host_to_dev(struct sm2_ep *ep,
```

Comment:
This name seems misleading. This sounds like you are processing a xfer_entry that had proto = ipc_host_to_dev which is not what you're doing here. This is where you send back the entry to fallback to the IPC call.